src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -157,20 +160,19 @@ public virtual async Task<T> GetConfigurationAsync(CancellationToken cancel)
             //   reach out to the metadata endpoint. Since multiple threads could be calling this method
             //   we need to ensure that only one thread is actually fetching the metadata.
             // else
-            //   if task is running, return the current configuration
+            //   if update task is running, return the current configuration
             //   else kick off task to update current configuration
             if (_currentConfiguration == null)
             {
-                await _configurationNullLock.WaitAsync(cancel).ConfigureAwait(false);
-                if (_currentConfiguration != null)
-                {
-                    _configurationNullLock.Release();
-                    return _currentConfiguration;
-                }
-
-#pragma warning disable CA1031 // Do not catch general exception types
                 try
                 {
+                    await _configurationNullLock.WaitAsync(cancel).ConfigureAwait(false);
+                    if (_currentConfiguration != null)
+                        return _currentConfiguration;
+
+                    Interlocked.Exchange(ref _configurationRetrieverState, ConfigurationRetrieverRunning);
+                    NumberOfTimesAutomaticRefreshRequested++;
```

Comment:
Confirming my understanding:

This doesn't need to be interlocked.increment because of the null lock above?

Reply:
Good call out. For the calling code it should do a Interlocked.Read, which we get for int64/uint64